src/common/tests_dmg_helpers.c

```diff
@@ -1332,8 +1332,7 @@ dmg_storage_device_list(const char *dmg_config_file, int *ndisks,
 			D_DEBUG(DB_TEST, "key1:\"%s\",val1=%s\n", key1,
 				json_object_to_json_string(val1));
 
-			json_object_object_get_ex(val1, "smd_info", &smd_info);
-			if (smd_info != NULL) {
+			if (json_object_object_get_ex(val1, "smd_info", &smd_info)) {
```

Comment:
you might trigger another coverity issue here of using smd_info as possible NULL. i know it will not be the case in reality, but coverity does not understand that. so you should probably add
D_ASSERT(smd_info != NULL);
after the if condition.

Reply:
We do expected the NULL in cases and we want to continue the loop so adding the else loop with continue. Hope this is ok.

Reply:
sorry i do not understand how you can expect NULL and the check before your change explicitly had:
if (smd_info != NULL) ..
for that code block.

How can you expect NULL and pass it to json_object_object_get_ex() on line 1337 ??

Reply:
We loop through JSON output for "host_storage_map" as key and value. We expected to have the First key as "storage" which has the smd_info with device details. Another key is hosts "hosts": "wolf-173:10001". For this key "hosts:, we do  not have any smd_info so it will return as NULL.
If I explicitly check for NULL and return error, test is going to fail unless we change the code and so does check for specific storage as key value only.

`  
  "response": {
    "host_errors": {},
    "host_storage_map": {
      "14815751108742920630": {
        "storage": {
          "nvme_devices": null,
          "scm_modules": null,
          "scm_namespaces": null,
          "scm_mount_points": null,
          "smd_info": {
            "devices": [
              {
                "role_bits": 0,
                "uuid": "5c3e7e5d-427b-4d43-8134-f94f291c817f",
                "tgt_ids": [
                  0,
                  1,
                  2,
                  3,
                  4,
                  5,
                  6,
                  7
                ],
                "rank": 0,
                "total_bytes": 0,
                "avail_bytes": 0,
                "usable_bytes": 0,
                "cluster_size": 0,
                "meta_size": 0,
                "meta_wal_size": 0,
                "rdb_size": 0,
                "rdb_wal_size": 0,
                "roles": "NA",
                "has_sys_xs": false,
                "ctrlr": {
                  "info": "",
                  "model": "INTEL SSDPE2MD800G4",
                  "serial": "CVFT6312000A800HGN",
                  "pci_addr": "0000:83:00.0",
                  "fw_rev": "8DV10171",
                  "vendor_id": "0x8086",
                  "pci_type": "",
                  "socket_id": 1,
                  "health_stats": null,
                  "namespaces": [
                    {
                      "id": 1,
                      "size": 800166076416
                    }
                  ],
                  "smd_devices": null,
                  "dev_state": "EVICTED",
                  "led_state": "NA"
                },
                "ctrlr_namespace_id": 1
              },
              {
                "role_bits": 0,
                "uuid": "0b9ba6bc-74b9-49a3-86d2-5c4a0c59858e",
                "tgt_ids": [
                  0,
                  1,
                  2,
                  3,
                  4,
                  5,
                  6,
                  7
                ],
                "rank": 1,
                "total_bytes": 0,
                "avail_bytes": 0,
                "usable_bytes": 0,
                "cluster_size": 0,
                "meta_size": 0,
                "meta_wal_size": 0,
                "rdb_size": 0,
                "rdb_wal_size": 0,
                "roles": "NA",
                "has_sys_xs": false,
                "ctrlr": {
                  "info": "",
                  "model": "INTEL SSDPE2MD800G4",
                  "serial": "CVFT63120001800HGN",
                  "pci_addr": "0000:84:00.0",
                  "fw_rev": "8DV10171",
                  "vendor_id": "0x8086",
                  "pci_type": "",
                  "socket_id": 1,
                  "health_stats": null,
                  "namespaces": [
                    {
                      "id": 1,
                      "size": 800166076416
                    }
                  ],
                  "smd_devices": null,
                  "dev_state": "NORMAL",
                  "led_state": "NA"
                },
                "ctrlr_namespace_id": 1
              }
            ],
            "pools": {}
          },
          "reboot_required": false,
          "mem_info": null
        },
        "hosts": "wolf-173:10001"
      }
    }
  },
  "error": null,
  "status": 0
}
`

Reply:
Im sorry, i am not familiar with this code. but i understand now that my original comment was wrong.

However the original code had:
json_object_object_get_ex(val1, "smd_info", &smd_info);
if (smd_info != NULL) {
xyz
}

you are doing more than a coverity fix with your change that does:
if (json_object_object_get_ex(val1, "smd_info", &smd_info)) {
xyz
}
because now xyz is being executed with possibly a NULL smd_info. 
this is what prompted my original comment.
to maintain the original behavior you need to change it such:

if (json_object_object_get_ex(val1, "smd_info", &smd_info)) {
  if(smd_info == NULL)
     continue;
  xyz;
}